vlayer/ethereum/circuit/lib/src/token.nr

```diff
@@ -0,0 +1,36 @@
+use dep::ethereum_history_api::{account_with_storage::{get_account_with_storage, StorageWithinBlock}, misc::types::{Address, Bytes32}};
+use dep::std::unsafe::zeroed;
+
+global TOKEN_BALANCE_INDEX = 0;
+
+struct Erc20Token {
+    address: Address,
+    balances_slot: Field,
+    chain_id: Field
+}
+
+impl Erc20Token {
+    fn calculate_balance_storage_key(self, wallet_address: Address) -> Bytes32 {
+        assert(false, "Not implemented");
+        zeroed()
+    }
+}
+
+trait TokenApi {
```

Comment:
I would call it simply ERC20 or IERC20 or ERC20Trait

IERC20 is used by OpenZeppelin